stix/idl/processing/subcollimator/stx_subc_transmission.pro

```diff
@@ -6,65 +6,91 @@
 ;
 ; PURPOSE:
 ;
-;   Compute the trasmission of a STIX subcollimator corrected for internal shadowing
+;   Compute the transmission of a STIX subcollimator corrected for internal shadowing
 ;
 ; CALLING SEQUENCE:
 ;
 ;   subc_transmission = stx_subc_transmission(flare_loc)
 ;
 ; INPUTS:
 ;
-;   flare_loc: bidimensional array containing the X and Y coordinate of the flare location 
+;   flare_loc: bidimensional array containing the X and Y coordinate of the flare location
 ;             (arcsec, in the STIX coordinate frame)
 ;
+;   ph_in: an array of photon energies [keV] (restricted to range 1-1000)
+;
 ; OUTPUTS:
 ;
 ;   A float number that represent the subcollimator transmission value
 ;
 ; HISTORY: August 2022, Massa P., first version (working only for detectors 3 to 10)
+;          11-Jul-2023, ECMD (Graz), updated following Recipe for STIX Flux and Amplitude Calibration (8-Nov 2022 gh)
+;                                    to include transparency and corner cutting for pure Tungsten grids
+;          24-Oct-2023, ECMD (Graz), added default to calculate low energy approximation if no input photon energies are passed                  
 ;
 ; CONTACT:
 ;   paolo.massa@wku.edu
 ;-
 
 
-function stx_subc_transmission, flare_loc
+function stx_subc_transmission, flare_loc, ph_in
+
+if ~keyword_set(ph_in) then begin
+  message, 'No photon energies passed, calculating low energy approximation at 1 keV.', /info
+  ph_in = 1.
+endif
+
+  restore,loc_file( 'grid_temp.sav', path = getenv('STX_GRID') )
+  fff=read_ascii(loc_file( 'grid_param_front.txt', path = getenv('STX_GRID') ),temp=grid_temp)
+  rrr=read_ascii(loc_file( 'grid_param_rear.txt', path = getenv('STX_GRID') ),temp=grid_temp)
+
+  transm = fltarr(n_elements(ph_in),32)
+
+  mass_attenuation = xsec(ph_in, 74,'AB',/cm2perg, error=error,use_xcom=1)
+  gmcm = 19.30
+
+  linear_attenuation = mass_attenuation*gmcm/10.
+
+  theta = sqrt( ( flare_loc[0]* flare_loc[0]) + ( flare_loc[1]* flare_loc[1]) ) / 3600./!radeg
```

Comment:
Can we rewrite this line as `theta = sqrt( flare_loc[0]^2. + flare_loc[1]^2. ) / 3600. * !dtor`? It's exactly the same, but  I think it's more clear

Reply:
That's more clear but in checking I think we don't ever use theta here so I've removed it altogether.